src/generateNumber.js

```diff
@@ -0,0 +1,23 @@
+'use strict';
+
+const { getRandomDigit } = require('./getRandomDigit');
+
+function generateNumber(length = 4) {
+  const uniques = [];
+
+  for (let i = 0; i < length; i++) {
+    let number = getRandomDigit();
+
+    while (uniques.includes(number)) {
+      number = getRandomDigit();
+    }
+
+    uniques.push(number);
+  }
+
```

Comment:
Now the solution is written correctly, but algorithmically it is not very optimal, because you create a loop (`while`) in the middle of the loop (`for`), instead we can do with one `while` loop with checking values for uniqueness:
```
const wantedNumberLength = 4;
function generateNumber() {
  const uniqueNumbers = [];

  while (uniqueNumbers.length < wantedNumberLength ) {
    const number = getRandomDigit();
    if(!uniqueNumbers.includes(number)) {
      uniqueNumbers.push(number);
    }
  }

  return uniqueNumbers.join('');
}
```